pkg/policymutation/policymutation.go

```diff
@@ -235,7 +235,7 @@ func GeneratePodControllerRule(policy kyverno.ClusterPolicy, log logr.Logger) (p
 
 	// scenario B
 	if controllers == "none" {
```

Comment:
Would it be cleaner to introduce a "canAutoGen" method that returns a true / false based on various checks? 

Seems a bit confusing that getControllers returns "none" when there are controllers in the match  / exclude....

Reply:
Updated the name, also returned a bool to indicate the result. We still need "controllers" as we use that to generate annotation JSON patches.